### PR TITLE
Fix mirror and fade threading crashes

### PR DESCRIPTION
**Pull request by SamGondelman**
[FB7623](https://highfidelity.fogbugz.com/f/cases/7623/CRASH-Crashes-using-mirror-or-spectator-cam-Bugplats) lists a bunch of crashes seemingly related to the secondary camera and fade.  This should hopefully fix them.

Test plan:
- Start interface in a domain with no spectator camera or mirror.  Open the console and enter `Window.takeSecondaryCameraSnapshot()`.  It should save a black 2048x1024 snapshot.
- Go somewhere with multiple mirrors and a bunch of other content (dev-content is good).  Move between the mirrors.  You shouldn't crash.
- Stand in front of a mirror, open the console, and enter `Window.takeSecondaryCameraSnapshot()`.  This snapshot should be identical to the image in the mirror.
- Get another person to test with.  Both of you activate your bubbles.  Extensively test moving between the mirrors while repeatedly colliding with each other's bubbles.  You shouldn't crash.

### CODE REVIEW

File: interface/src/SecondaryCamera.cpp [hunk 1]
```diff
@@ -123,16 +128,10 @@ void SecondaryCameraJobConfig::enableSecondaryCameraRenderConfigs(bool enabled)
...
 
-void SecondaryCameraJobConfig::resetSizeSpectatorCamera(int width, int height) { // Carefully adjust the framebuffer / texture.
-    qApp->getRenderEngine()->getConfiguration()->getConfig<SecondaryCameraRenderTask>()->resetSize(width, height);
-}
-
-void SecondaryCameraRenderTaskConfig::resetSize(int width, int height) { // FIXME: Add an arg here for "destinationFramebuffer"
-    bool wasEnabled = isEnabled();
-    setEnabled(false);
-    auto textureCache = DependencyManager::get<TextureCache>();
-    textureCache->resetSpectatorCameraFramebuffer(width, height); // FIXME: Call the correct reset function based on the "destinationFramebuffer" arg
```
Comment (SamGondelman) [inline interface/src/SecondaryCamera.cpp]: The secondary camera framebuffer resizing was previously done from the script thread that called it, which could reset the textures in the middle of rendering.

File: interface/src/SecondaryCamera.cpp [hunk 2]
```diff
@@ -55,13 +46,15 @@ class SecondaryCameraJob {  // Changes renderContext for our framebuffer and vie
...
+        _textureWidth = config.textureWidth;
+        _textureHeight = config.textureHeight;
     }
 
     void run(const render::RenderContextPointer& renderContext, RenderArgsPointer& cachedArgs) {
         auto args = renderContext->args;
         auto textureCache = DependencyManager::get<TextureCache>();
         gpu::FramebufferPointer destFramebuffer;
-        destFramebuffer = textureCache->getSpectatorCameraFramebuffer(); // FIXME: Change the destination based on some unimplemented config var
+        destFramebuffer = textureCache->getSpectatorCameraFramebuffer(_textureWidth, _textureHeight); // FIXME: Change the destination based on some unimplemented config var
```
Comment (SamGondelman) [inline interface/src/SecondaryCamera.cpp]: Resizing is now handled here.

File: libraries/model-networking/src/model-networking/TextureCache.cpp [hunk 3]
```diff
@@ -1016,6 +1017,7 @@ NetworkTexturePointer TextureCache::getResourceTexture(QUrl resourceTextureUrl)
         if (_hmdPreviewFramebuffer) {
             texture = _hmdPreviewFramebuffer->getRenderBuffer(0);
             if (texture) {
+                texture->setSource(HMD_PREVIEW_FRAME_URL.toString().toStdString());
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: This and the above don't really do anything but they're helpful for debugging.

File: libraries/render-utils/src/FadeEffect.cpp [hunk 4]
```diff
@@ -29,11 +28,15 @@ void FadeEffect::build(render::Task::TaskConcept& task, const task::Varying& edi
...
-render::ShapePipeline::BatchSetter FadeEffect::getBatchSetter() const {
+render::ShapePipeline::BatchSetter FadeEffect::getBatchSetter() {
     return [this](const render::ShapePipeline& shapePipeline, gpu::Batch& batch, render::Args*) {
         auto program = shapePipeline.pipeline->getProgram();
         auto maskMapLocation = program->getTextures().findLocation("fadeMaskMap");
         auto bufferLocation = program->getUniformBuffers().findLocation("fadeParametersBuffer");
+        if (!_maskMap) {
+            auto texturePath = PathUtils::resourcesPath() + "images/fadeMask.png";
+            _maskMap = DependencyManager::get<TextureCache>()->getImageTexture(texturePath, image::TextureUsage::STRICT_TEXTURE);
+        }
```
Comment (SamGondelman) [inline libraries/render-utils/src/FadeEffect.cpp]: I'm not 100% sure about this, but it seems like moving the _maskMap creation out of the constructor and into this lambda fixed an issue where _maskMap has a junk value during rendering.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3657/HighFidelity-Beta-PR11466-49512c95d753aaf94209c551354c48b548d76490.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3657/HighFidelity-Beta-PR11466-49512c95d753aaf94209c551354c48b548d76490.exe

File: libraries/render-utils/src/FadeEffect.cpp (hunk 4 above)
Comment (samcake) [reply libraries/render-utils/src/FadeEffect.cpp]: Not a big fan of that
i don;t think that is the right fix
plus removing the const on the BatchSetter doesn;t seem like the right thing to do

**Comment by SamGondelman** [comment]
Ugh now I'm seeing the crash again.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3681/HighFidelity-Beta-PR11466-553829f7ab94acbd08c5dc97d255b0bede8cd77d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3681/HighFidelity-Beta-PR11466-553829f7ab94acbd08c5dc97d255b0bede8cd77d.exe

**Comment by conklin94122** [comment]
QA PASS

File: interface/src/Application.cpp [hunk 5]
```diff
@@ -682,6 +684,8 @@ bool setupEssentials(int& argc, char** argv, bool runningMarkerExisted) {
     DependencyManager::set<Ledger>();
     DependencyManager::set<Wallet>();
 
+    DependencyManager::set<FadeEffect>();
```
Comment (samcake) [inline interface/src/Application.cpp]: I hate that but ok....

**Pull request merged**